Apple M1 WDT

pipcet opened the pull request:
This adds support for the watchdog timer, which is also the standard way of rebooting these systems.

pipcet committed:
watchdog: apple_wdt: Add support for Apple M1 WDT

This is the watchdog timer on Apple M1 systems. It is the standard way
of rebooting these systems.

Signed-off-by: Pip Cet <pipcet@gmail.com>

pipcet commented:
(I've deliberately opened this in my own repository, as I'm currently working on the M1 machine and don't have a second machine handy for testing. Will submit to https://github.com/AsahiLinux/linux for @svenpeter42 et al to review once this is somewhat tested.)

svenpeter42 commented:
feel free to directly open this on the main repo, just mention that it hasn't been tested. i think we should try to iterate quite a bit faster here and accept that we'll break some stuff every now and then

pipcet committed:
dt-bindings: watchdog: bindings for Apple M1 WDT

This is the watchdog timer on Apple M1 systems. It is the standard way
of rebooting these systems.

Signed-off-by: Pip Cet <pipcet@gmail.com>

pipcet committed:
arm64: apple: Add WDT to devicetree

This is the watchdog timer on Apple M1 systems. It is the standard way
of rebooting these systems.

Signed-off-by: Pip Cet <pipcet@gmail.com>
